CSharp/Mazes/Mazes/Program.cs

```diff
@@ -22,31 +22,64 @@ class Program
         const int PEN_THICKNESS_IN_PIXLES = 3;
         static void Main(string[] args)
         {
+
+            // Prompt the user to input the type of algorithm we will be using. 1 for Binary 2 for sidewinder
+            System.Console.WriteLine("Please tell me what type of algorithm you want to use. (1 - Binary 2 - Sidewinder)");
+            int algoType;
+
+            // Create the base grid
             IGrid grid = new Grid.Grid(ROWS, COLUMNS);
-            IMazeAlgorithm binaryAlgorithm = new Binary();
+            // Determine and create the type of algorithm that will be used to generate the rooms
+            IMazeAlgorithm currentAlgorithm;
+            do
+            {
+                // Get the type of algorithm that the user wants to use
+                algoType = Convert.ToInt32(System.Console.ReadLine());
+
+                switch (algoType)
+                {
+                    case 1:
+                        currentAlgorithm = new Binary();
+                        break;
+
+                    case 2:
+                        currentAlgorithm = new Sidewinder();
+                        break;
+
+                    default:
+                        System.Console.WriteLine("Can't find this algorithm. Please choose a valid one. (1 - Binary 2 - Sidewinder)");
+                        currentAlgorithm = null;
```

Comment:
NIT: Make currentAlgortihm null when you declare it, so you don't have to re-declare it here.